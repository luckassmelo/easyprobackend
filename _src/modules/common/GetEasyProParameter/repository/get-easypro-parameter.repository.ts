import { IGetEasyPROParameterRepository } from '../interfaces/get-easypro-parameter-repository.interface';
import PostgresSQLAdapter from "../../../../../src/infra/database/PostgreSQLAdapter";
import { ParameterResponseProps } from '../types/param.types';

export class GetEasyPROParameterRepository implements IGetEasyPROParameterRepository {

  constructor(
    private readonly adapter: PostgresSQLAdapter
  ) { }

  async getParameter(idSite: number, name: string): Promise<ParameterResponseProps[]> {
    const response = await this.adapter
      .connection
      .select<ParameterResponseProps[]>("id_param_name as idParamName",
        "param_value as paramValue",
        "param_value_aux as paramValueAux",
        "param_value_default as paramValueDefault",
        "id_site as idSite",
        "server_resource as serverResource",
        "desc_resource as descResource")
      .from("param.tbl_param")
      .where("id_param_name", "=", name)
      .where("id_site", "=", idSite)
      .innerJoin("param.tbl_param_value", "tbl_param.id_param", "tbl_param_value.id_param")
      .innerJoin("param.tbl_resource_info", "tbl_param.id_resource", "param.tbl_resource_info.id_resource")
      .innerJoin("param.tbl_param_type", "tbl_param.id_param_type", "tbl_param_type.id_param_type")
      .innerJoin("param.tbl_resources", "tbl_param.id_resource", "tbl_resources.id_resource");

    return response;
  }

}
