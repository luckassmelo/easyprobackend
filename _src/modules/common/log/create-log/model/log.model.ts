import { ParameterNotFoundError } from '../../../../../infra/api/errors/parameter-not-found.error';
import { ParameterWrongTypeError } from '../../../../../infra/api/errors/parameter-wrong-type.error';


export class LogModel {
  private _processName: string;
  private _infoLog: any;

  public set processName(processName: string) {

    if (typeof processName !== 'string') {
      throw new ParameterWrongTypeError('processName', 'string');

    }
    if (!processName) {
      throw new ParameterNotFoundError('processName');
    }

    this._processName = processName;
  }

  public set infoLog(infoLog: any) {
    if (!infoLog || (Object.keys(infoLog).length === 0 && typeof infoLog === 'object')) {
      throw new ParameterNotFoundError('infoLog');
    }
    if (typeof infoLog !== 'object') {
      throw new ParameterWrongTypeError('infoLog', 'object');
    }

    this._infoLog = infoLog;
  }

  public get processName(): string {
    return this._processName;
  }

  public get infoLog(): any {
    return this._infoLog;
  }
}
