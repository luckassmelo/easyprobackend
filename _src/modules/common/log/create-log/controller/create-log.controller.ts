import { Controller } from '../../../../../presentation/protocols/controller';
import { HttpResponse } from '../../../../../presentation/protocols/http';
import { LogModel } from '../model/log.model';
import { CreateLogUseCase } from '../usecase/create-log.usecase';

type CreateLogProps = {
  processName: string;
  infoLog: any;
};

export class CreateLogController implements Controller {

  constructor(private usecase: CreateLogUseCase) { }

  async handle(props: CreateLogProps): Promise<HttpResponse> {

    const { processName, infoLog } = props;

    const log = new LogModel();
    log.processName = processName;
    log.infoLog = infoLog;

    const response = await this.usecase.execute(log);

    return {
      statusCode: 200,
      response: response,
      message: `Success with creation of log.`
    }
  }

}
