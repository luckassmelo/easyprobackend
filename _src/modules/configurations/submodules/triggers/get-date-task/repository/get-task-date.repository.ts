import { IGetTaskDate } from "../interface/get-task-date.interface";
import PostgresSQLAdapter from '../../../../../../../src/infra/database/PostgreSQLAdapter'
import { ResponseDateType } from "../types/response-date.type";
import { GetDateOfTask } from "../models/get-date.model";


export class GetDateOfTaskRepository implements IGetTaskDate {

    constructor(private readonly adapter: PostgresSQLAdapter){}

    async findDateById(id: GetDateOfTask): Promise<ResponseDateType> {
        const response = await this.adapter.connection('trigger.tbl_trigger_task')
        .select<ResponseDateType>('created_at')
        .where('id_trigger', '=', id.getId())
        .orderBy('created_at', 'desc')
        .limit(1);


        return response[0];
    }
}