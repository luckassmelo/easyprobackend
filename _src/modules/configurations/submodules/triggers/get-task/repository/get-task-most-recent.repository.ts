import { IGetTaskDate } from "../interface/get-task-most-recent.interface";
import PostgresSQLAdapter from '../../../../../../../src/infra/database/PostgreSQLAdapter'
import {  GetTriggerTask} from "../models/get-task-most-recent.model";


export class GetDateOfTaskRepository implements IGetTaskDate {

    constructor(private readonly adapter: PostgresSQLAdapter){}

    async findDateById(props:  GetTriggerTask): Promise< GetTriggerTask> {
        const response = await this.adapter.connection('trigger.tbl_trigger_task')
        .select<GetTriggerTask>('*')
        .where('id_trigger', '=', props.idTrigger).andWhere('id_oee', '=', props.idOee)
        .orderBy('created_at', 'desc')
        .limit(1);

        return response;
    }
}