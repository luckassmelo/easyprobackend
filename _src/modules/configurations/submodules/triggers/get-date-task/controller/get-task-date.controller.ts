import { GetTaskDateUseCase } from "../usecases/get-task-date.usecase";
import {Controller} from '../../../../../../presentation/protocols/controller';
import { HttpResponse } from "../../../../../../presentation/protocols/http";
import {  GetTriggerTask} from "../models/get-date.model";

type RequestProps = {
    id: number
}

export class GetTaskDateController implements Controller{
    constructor(private usecase: GetTaskDateUseCase){}

    async handle (request: RequestProps): Promise<HttpResponse>{
        const { id }  = request;
        
        const getTriggerTask = new GetTriggerTask()
        getTriggerTask.idTriggerSet = Number(id)
        const getDate =  await this.usecase.execute(getTriggerTask);
        
            
        return {
            statusCode: 200,
            response: getDate
        };
    };
}