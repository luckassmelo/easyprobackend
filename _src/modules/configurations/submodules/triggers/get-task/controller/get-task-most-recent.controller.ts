import { GetTaskDateUseCase } from "../usecases/get-task-most-recent.usecase";
import {Controller} from '../../../../../../presentation/protocols/controller';
import { HttpResponse } from "../../../../../../presentation/protocols/http";
import {  GetTriggerTask} from "../models/get-task-most-recent.model";

type RequestProps = {
    idTrigger: number
    idOee: number
}

export class GetTaskDateController implements Controller{
    constructor(private usecase: GetTaskDateUseCase){}

    async handle (request: RequestProps): Promise<HttpResponse>{
        const { idTrigger, idOee }  = request;
        
        const getTriggerTask = new GetTriggerTask()
        getTriggerTask.idTrigger = Number(idTrigger);
        getTriggerTask.idOee = Number(idOee);
        
        const getDate =  await this.usecase.execute(getTriggerTask);
        
            
        return {
            statusCode: 200,
            response: getDate
        };
    };
}