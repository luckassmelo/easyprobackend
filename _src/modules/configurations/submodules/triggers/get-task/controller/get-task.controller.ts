import { GetTaskDateUseCase } from "../usecases/get-task.usecase";
import {Controller} from '../../../../../../presentation/protocols/controller';
import { HttpResponse } from "../../../../../../presentation/protocols/http";
import {  GetTriggerTask} from "../models/get-task.model";

type RequestProps = {
    id: number
}

export class GetTaskDateController implements Controller{
    constructor(private usecase: GetTaskDateUseCase){}

    async handle (request: RequestProps): Promise<HttpResponse>{
        const { id }  = request;
        
        const getTriggerTask = new GetTriggerTask()
        getTriggerTask.idTrigger = Number(id)
        const getDate =  await this.usecase.execute(getTriggerTask);
        
            
        return {
            statusCode: 200,
            response: getDate
        };
    };
}