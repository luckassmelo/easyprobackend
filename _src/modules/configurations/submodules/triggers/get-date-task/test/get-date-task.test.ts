import {GetTaskDateController} from '../controller/get-task-date.controller';
import { GetDateOfTaskRepository } from '../repository/get-task-date.repository';
import { GetTaskDateUseCase } from '../usecases/get-task-date.usecase';
import {connectionProductionManager} from '../../../../../../../src/infra/database/index';
import { GetDateOfTask } from '../models/get-date.model';
import { ParameterNotFoundError } from '../../../../../../infra/api/errors/parameter-not-found.error';
import { CustomError } from '../../../../../../domain/errors/custom.error';

afterAll(async() =>{
    await connectionProductionManager.close();
})
describe('Unit tests for get the last date of task', ()=>{
    const getDateTaskRepository =  new GetDateOfTaskRepository(connectionProductionManager);
    const getDateTaskUseCase = new GetTaskDateUseCase(getDateTaskRepository);
    const getDateTaskController = new GetTaskDateController(getDateTaskUseCase);
    const TIMEOUT_IN_MILLISECONDS = 10000;

    test('should return the last date of task', async ()=>{
        const dateReturned =  await getDateTaskUseCase.execute(new GetDateOfTask({
            id: '2'
        }));

        expect(dateReturned).not.toBeNull();
    }, TIMEOUT_IN_MILLISECONDS);

    test('should return an error when id is not providade', async() =>{
        try{
            await getDateTaskController.handle({
                id: ''
            });
            fail('Expected an error to be thrown');
        } catch(error){
            expect(error).toBeInstanceOf(CustomError);
            expect(error.message).toBe('Parameter id not found')
        }
        
    }, TIMEOUT_IN_MILLISECONDS);
})