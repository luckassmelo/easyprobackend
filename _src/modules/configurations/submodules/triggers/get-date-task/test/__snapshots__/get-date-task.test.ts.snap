// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for get the last date of task should return an error when id is not providade: RequestError: Parameter id not found 1`] = `"Parameter id not found"`;
